dvc/remote/local/__init__.py

```diff
@@ -566,3 +593,88 @@ def unprotect(path_info):
     @staticmethod
     def protect(path_info):
         os.chmod(path_info.path, stat.S_IREAD | stat.S_IRGRP | stat.S_IROTH)
+
+    @classmethod
+    def _append_unpacked_suffix(cls, string):
+        return string + cls.UNPACKED_DIR_SUFFIX
+
+    def _get_unpacked_dir_info(self, checksum):
+        unpacked_dir_info = self.checksum_to_path_info(checksum)
+        unpacked_dir_info.path = self._append_unpacked_suffix(
+            unpacked_dir_info.path
+        )
+
+        return unpacked_dir_info
+
+    def _update_unpacked_dir(self, checksum, dir_info):
+        unpacked_dir_info = self._get_unpacked_dir_info(checksum)
+
+        exists = self.exists(unpacked_dir_info)
+        if exists:
+            if self._path_metadata_changed(unpacked_dir_info.path):
```

Comment:
why not just `if not self.state.get()`? You've `self.state.save(unpacked_dir_info, checksum)` earlier.